### ISSUE TITLE
linstor node info returns errors

### ISSUE THREAD

**Issue by acidrop** [LINBIT/linstor-client#47]
I see the following output when issuing 'linstor n info' ... is this a bug?
Output shows some info only for pve4 and pve5 nodes, but none for pve1-3.
Using latest versions of linstor packages.

╭─────────────────────────────────────────────────────╮
┊ Node ┊ NodeType  ┊ Addresses               ┊ State  ┊
╞═════════════════════════════════════════════════════╡
┊ pve1 ┊ COMBINED  ┊ 10.10.12.1:3366 (PLAIN) ┊ Online ┊
┊ pve2 ┊ COMBINED  ┊ 10.10.12.2:3366 (PLAIN) ┊ Online ┊
┊ pve3 ┊ COMBINED  ┊ 10.10.12.3:3366 (PLAIN) ┊ Online ┊
┊ pve4 ┊ SATELLITE ┊ 10.10.12.4:3366 (PLAIN) ┊ Online ┊
┊ pve5 ┊ SATELLITE ┊ 10.10.12.5:3366 (PLAIN) ┊ Online ┊
╰─────────────────────────────────────────────────────╯

╭────────────────────────────────────────────────────────╮
┊ Node ┊ Diskless ┊ LVM ┊ LVMThin ┊ ZFS/Thin ┊ File/Thin ┊
╞════════════════════════════════════════════════════════╡
┊ pve4 ┊ +        ┊ +   ┊ +       ┊ +        ┊ +         ┊
┊ pve5 ┊ +        ┊ +   ┊ +       ┊ +        ┊ +         ┊
╰────────────────────────────────────────────────────────╯
Unsupported storage providers:
 pve4:
  SPDK: IO exception occured when running 'rpc.py get_spdk_version': Cannot run program "rpc.py": error=2, No such file or directory
  EXOS: IO exception occured when running 'lsscsi --version': Cannot run program "lsscsi": error=2, No such file or directory
  EXOS: '/bin/bash -c cat /sys/class/sas_phy/*/sas_address' returned with exit code 1
  EXOS: '/bin/bash -c cat /sys/class/sas_device/end_device-*/sas_address' returned with exit code 1
 pve5:
  SPDK: IO exception occured when running 'rpc.py get_spdk_version': Cannot run program "rpc.py": error=2, No such file or directory
  EXOS: IO exception occured when running 'lsscsi --version': Cannot run program "lsscsi": error=2, No such file or directory
  EXOS: '/bin/bash -c cat /sys/class/sas_phy/*/sas_address' returned with exit code 1
  EXOS: '/bin/bash -c cat /sys/class/sas_device/end_device-*/sas_address' returned with exit code 1

╭─────────────────────────────────────╮
┊ Node ┊ DRBD ┊ LUKS ┊ NVMe ┊ Storage ┊
╞═════════════════════════════════════╡
┊ pve4 ┊ +    ┊ +    ┊ +    ┊ +       ┊
┊ pve5 ┊ +    ┊ +    ┊ -    ┊ +       ┊
╰─────────────────────────────────────╯
Unsupported resource layers:
 pve5:
  NVME: IO exception occured when running 'nvme version': Cannot run program "nvme": error=2, No such file or directory
  OPENFLEX: IO exception occured when running 'nvme version': Cannot run program "nvme": error=2, No such file or directory

**Comment by rp-** [comment]
can you provide the output of
`curl http://localhost:3370/v1/nodes`

**Comment by rck** [comment]
very wild guessing here, but the ones that don't show up are 'COMBINED', Rene, might that be the problem and they get "filtered"?

**Comment by acidrop** [comment]
[{"name":"pve1","type":"COMBINED","props":{"CurStltConnName":victAllowEviction":"false"},"net_interfaces":[{"name":"defaul,"satellite_encryption_type":"PLAIN","is_active":true,"uuid":ction_status":"ONLINE","uuid":"e57bcd4b-8d14-4f49-ba17-c4f78fLVM_THIN","ZFS","ZFS_THIN","FILE","FILE_THIN","OPENFLEX_TARGEITECACHE","CACHE","OPENFLEX","STORAGE"],"unsupported_provider'rpc.py get_spdk_version': Cannot run program \"rpc.py\": errxception occured when running 'lsscsi --version': Cannot run rectory","'/bin/bash -c cat /sys/class/sas_phy/*/sas_address'/sys/class/sas_device/end_device-*/sas_address' returned withD","props":{"CurStltConnName":"default","NodeUname":"pve2","Dt_interfaces":[{"name":"default","address":"10.10.12.2","sateLAIN","is_active":true,"uuid":"ec0acf4a-e350-4dcf-b2c5-9a43114bb56c16-ed00-4c3b-b825-b748d5a181fc","storage_providers":["DE","FILE_THIN","OPENFLEX_TARGET"],"resource_layers":["DRBD","TORAGE"],"unsupported_providers":{"SPDK":["IO exception occurot run program \"rpc.py\": error=2, No such file or directorylsscsi --version': Cannot run program \"lsscsi\": error=2, Nos/class/sas_phy/*/sas_address' returned with exit code 1","'/e-*/sas_address' returned with exit code 1"]}},{"name":"pve3"default","NodeUname":"pve3","DrbdOptions/AutoEvictAllowEvicti","address":"10.10.12.3","satellite_port":3366,"satellite_enc1a6c2d0d-e09c-4970-9ac5-ed95a8642832"}],"connection_status":"8c203","storage_providers":["DISKLESS","LVM","LVM_THIN","ZFS""],"resource_layers":["DRBD","LUKS","NVME","WRITECACHE","CACH":{"SPDK":["IO exception occured when running 'rpc.py get_spdr=2, No such file or directory"],"EXOS":["IO exception occurerogram \"lsscsi\": error=2, No such file or directory","'/binreturned with exit code 1","'/bin/bash -c cat /sys/class/sas_exit code 1"]}},{"name":"pve4","type":"SATELLITE","props":{"CrbdOptions/AutoEvictAllowEviction":"false"},"net_interfaces":llite_port":3366,"satellite_encryption_type":"PLAIN","is_acti0c8c0d"}],"connection_status":"ONLINE","uuid":"b458413a-b880-ISKLESS","LVM","LVM_THIN","ZFS","ZFS_THIN","FILE","FILE_THIN"LUKS","NVME","WRITECACHE","CACHE","OPENFLEX","STORAGE"],"unsued when running 'rpc.py get_spdk_version': Cannot run program"],"EXOS":["IO exception occured when running 'lsscsi --versi such file or directory","'/bin/bash -c cat /sys/class/sas_phbin/bash -c cat /sys/class/sas_device/end_device-*/sas_addres,"type":"SATELLITE","props":{"CurStltConnName":"default","Nodion":"false"},"net_interfaces":[{"name":"default","address":"cryption_type":"PLAIN","is_active":true,"uuid":"7d6bac12-ab8f"ONLINE","uuid":"7a9cc769-ffde-4371-a46f-586fc0176c05","stora","ZFS_THIN","FILE","FILE_THIN","OPENFLEX_TARGET"],"resource_ORAGE"],"unsupported_providers":{"SPDK":["IO exception occuret run program \"rpc.py\": error=2, No such file or directory"sscsi --version': Cannot run program \"lsscsi\": error=2, No /class/sas_phy/*/sas_address' returned with exit code 1","'/b-*/sas_address' returned with exit code 1"]},"unsupported_layg 'nvme version': Cannot run program \"nvme\": error=2, No sun occured when running 'nvme version': Cannot run program \"n

**Comment by rp-** [comment]
> very wild guessing here, but the ones that don't show up are 'COMBINED', Rene, might that be the problem and they get "filtered"?

yep, you are right
will be fixed soon :tm:

**Comment by acidrop** [comment]
Thanks Rene for fixing on linstor-client  v1.8.0. Just a quick question related to this. Is it possible to suppress these kind of messages when issuing 'linstor node info':

```
SPDK: IO exception occured when running 'rpc.py get_spdk_version': Cannot run program "rpc.py": error=2, No such file or directory
EXOS: '/bin/bash -c cat /sys/class/sas_phy/*/sas_address' returned with exit code 1
EXOS: '/bin/bash -c cat /sys/class/sas_device/end_device-*/sas_address' returned with exit code 1
```

I'm not using EXOS neither SPDK, not sure if this related to a missing package on my systems ?

Thanks

**Comment by rp-** [comment]
No there is no way to suppress it, it rather should just show why our check failed to support those features.
I mean if it is really so important for you to hide those check messages I can add an option to hide them

**Comment by acidrop** [comment]
In my opinion, there are 3 reasons:

1. Aesthetically, does not look good having any kind of warning errors on the screen.
2. The error/warnings can be misleading, thinking that there's something wrong going on (initially that's what I thought).
3. When you just execute 'linstor node info" (without the "-n" parameter) and you have >3 nodes, then the screen will fill up with those warning/errors, possibly hiding the real information.

If the list of "Unsupported storage providers" cannot be hidden by default, then yes I think an option to hide them would be good to have.

These are all just my personal opinion which might be wrong :)

**Comment by rp-** [comment]
Fixed in v1.9.0
